aten/src/ATen/TensorIndexing.h

```diff
@@ -37,52 +37,48 @@ TORCH_API extern const EllipsisIndexType Ellipsis;
 
 struct TORCH_API Slice final {
  public:
-  // This mirrors `__PySlice_Unpack` in torch/csrc/utils/python_compat.h
   Slice(
-      c10::optional<int64_t> start_index = c10::nullopt,
-      c10::optional<int64_t> stop_index = c10::nullopt,
-      c10::optional<int64_t> step_index = c10::nullopt) {
+      c10::optional<c10::SymInt> start_index = c10::nullopt,
+      c10::optional<c10::SymInt> stop_index = c10::nullopt,
+      c10::optional<c10::SymInt> step_index = c10::nullopt) {
     if (!step_index.has_value()) {
-      step_ = 1;
+      step_ = c10::SymInt(1);
     } else {
       step_ = step_index.value();
```

Comment:
Now that these are a symints, they aren't trivially copyable so you should move them when you are done with them.
```suggestion
    step_ = std::move(step_index).value();
```